pkg/apis/kubeone/v1beta2/helpers.go

```diff
@@ -44,6 +44,8 @@ func SetCloudProvider(cp *CloudProviderSpec, name string) error {
 		cp.Openstack = &OpenstackSpec{}
 	case "equinixmetal", "packet":
 		cp.EquinixMetal = &EquinixMetalSpec{}
+	case "vmware-cloud-director":
```

Comment:
Same here -- shouldn't we have the JSON tag (`vmwareCloudDirector`) here?

Reply:
This is the expected value.

Reply:
This is used in Terraform, we can actually change it to `vmwareCloudDirector` to be in sync. @kron4eg WDYT?

Reply:
Done